test/component2_suggestors/deprecated_lifecycle_suggestor_test.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/component2_suggestors/deprecated_lifecycle_suggestor.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+import 'package:over_react_codemod/src/component2_suggestors/component2_constants.dart';
+
+main() {
+  group('DeprecatedLifecycleSuggestor', () {
+    final testSuggestor = getSuggestorTester(DeprecatedLifecycleSuggestor());
+
+    test('empty file', () {
+      testSuggestor(expectedPatchCount: 0, input: '');
+    });
+
+    test('no matches', () {
+      testSuggestor(
+        expectedPatchCount: 0,
+        input: '''
+          library foo;
+          var a = 'b';
+          class Foo {}
+        ''',
+      );
+    });
+
+    test('adds a FIXME comment for componentWillUpdate with override', () {
```

Comment:
Just like i said elsewhere, especially if we are going to add `componentWillMount` to this codemod you could consolidate these tests into a single shared test suite.

```
void sharedDeprecationFixmeCommentTests(String methodName){
      test('adds a FIXME comment for $methodName with override', () {
        testSuggestor(
          expectedPatchCount: 1,
          input: '''
            @Component2()
            class FooComponent extends UiComponent2 {
                @override
                $methodName(){}
            }
          ''',
          expectedOutput: '''
            @Component2()
            class FooComponent extends UiComponent2 {
                ${getDeprecationWithRevertCommentForLifecycleMethod(methodName)}
                @override
                $methodName(){}
            }
          ''',
        );
      });

      test('adds a FIXME comment for $methodName without override', () {
        testSuggestor(
          expectedPatchCount: 1,
          input: '''
            @Component2()
            class FooComponent extends UiComponent2 {
                $methodName(){}
            }
          ''',
          expectedOutput: '''
            @Component2()
            class FooComponent extends UiComponent2 {
                ${getDeprecationWithRevertCommentForLifecycleMethod(methodName)}
                $methodName(){}
            }
          ''',
        );
      });
    }

    deprecatedLifecycleMethods.forEach((methodName){
      sharedDeprecationFixmeCommentTests(methodName);
    });

    test('adds all FIXME comments when $deprecatedLifecycleMethods are all present', () {
      testSuggestor(
        expectedPatchCount: deprecatedLifecycleMethods.length,
        input: '''
          @Component2()
          class FooComponent extends UiComponent2 {
              ${deprecatedLifecycleMethods.map((methodName) {
                return '@override\n${methodName}(){}';
              }).join('\n')}
          }
        ''',
        expectedOutput: '''
          @Component2()
          class FooComponent extends UiComponent2 {
              ${deprecatedLifecycleMethods.map((methodName) {
                return '${getDeprecationWithRevertCommentForLifecycleMethod(methodName)}\n@override\n${methodName}(){}';
              }).join('\n')}
          }
        ''',
      );
    });
```